scripts/generate_user_permissions.py

```diff
@@ -0,0 +1,117 @@
+    #   "create":"allowed", "update":"allowed", "delete":"allowed", 
+    #   "view":"allowed", "killTask":"allowed"
+    #
+    #-------------------------------------------------------------------------------    
+
+def getUserPermissionRows(sheet_titles, oauth_credentials_file):
+    scope = ['https://spreadsheets.google.com/feeds']
+    # Currently, the email defined under this file has to have edit permissions on the file
+    # todo : check if we can get rid of that and use our email instead of the generated
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(oauth_credentials_file, scope)
+    gc = gspread.authorize(credentials)
+
+    # open by title - titles can change and so is fragile, keys don't
+    # this works as well
+    # gc.open_by_key("1Izpq1eZI1NuMBddpvvxsIt3usWm11j56eJwPdfxCw6E").sheet1
+    rows = []
+    for title in sheet_titles:
+        sheet = gc.open(title).sheet1
+        rows = rows + sheet.get_all_records() 
+    return rows
+    #-------------------------------------------------------------------------------
+
+
+def generateUserPermissionsJson(rows, password_column):
+    multiple_user_permissions_per_user = {}
+    user_permissions_json = []
+    actions = ["create", "update", "delete", "view", "killTask"]
+    permissions_json = {}
+    credentials = {}
+
+    for row in rows:
+        # todo : "username" and "path" are required keys, bailout if they are not defined
+        # dictionary with keys username, team, path, create, update, delete, view, killTask
+        actions_allowed = []
+        user = row["username"]
+        if (row["path"] == None or row["path"] == ""):
+            continue
+        # remove spaces from key and values 
+        row = {k.translate(None, ' '): v.translate(None, ' ') for k, v in row.iteritems()}
+        if any ( action in row.keys() for action in actions):
+            # user custom permissions        
+            if ("create" in row.keys() and row["create"] == "allowed"): actions_allowed.append("create")
+            if ("update" in row.keys() and row["update"] == "allowed"): actions_allowed.append("update")
+            if ("delete" in row.keys() and row["delete"] == "allowed"): actions_allowed.append("delete")
+            if ("view" in row.keys() and row["view"] == "allowed") : actions_allowed.append("view")
+            if ("killTask" in row.keys() and row["killTask"] == "allowed"): actions_allowed.append("killTask")
+        else:
+            actions_allowed = actions
+
+        for action in actions_allowed:    
+            permission = {}
+            permission["allowed"] = action
+            permission["on"] = row["path"]
+            if (user not in permissions_json.keys()):
+                permissions_json[user] = []
+            permissions_json[user].append(permission) 
+
+        if (credentials.get(user) == None or credentials.get(user) == ""):
+            if (password_column in row.keys()):
```

Comment:
made it optional but defaulting to username so removed the check 

:white_check_mark: